src/components/TimePeriodFilter.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState } from 'react';
+import cx from "classnames";
+
+export interface TimePeriodFilterProps {
```

Comment:
I think if we wanted to simplify this a little bit we could just pass two props to `TimePeriodFilter`:

```suggestion
export interface TimePeriodFilterProps {
  aggPeriod: string;
  onAggPeriodChanged: (value: string) => void;
}
```

I think the parent can derive the other two values from the `aggPeriod`, and this allows this component to behave a bit more like a controlled component